### PR TITLE
Run sandbox as ENV defined user

### PR DESCRIPTION
**Pull request by pivnicek**
This should run the sandbox under the environment supplied NB_USER

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -102,15 +136,14 @@ RUN jupyter nbextension enable --py widgetsnbextension --sys-prefix \
...
 RUN MPLBACKEND=Agg python -c "import matplotlib.pyplot"
 
-USER root
-
-### Custom stuff 
 
+### Custom stuff
 # Install KBC Transformation package
-RUN pip3 install --no-cache-dir --upgrade git+git://github.com/keboola/python-transformation.git@1.1.13
+RUN pip3 install --no-cache-dir --upgrade --user git+git://github.com/keboola/python-transformation.git@1.1.13
```
Comment (odinuv) [inline Dockerfile]: run that under root

File: Dockerfile [hunk 2]
```diff
@@ -94,6 +126,8 @@ RUN pip3 install --no-cache-dir \
     xlrd \
     qgrid
 
+# RUN fix-permissions /etc/jupyter
```
Comment (odinuv) [inline Dockerfile]: del or uncomment?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
       - KBC_ROW_ID
       - KBC_CONFIG_VERSION
       - KBC_TOKEN
+      - KBC_STORAGEAPI_URL
   sandbox:
-    build: .
+    build:
+      context: .
+      args:
+        NB_USER: ${NB_USER}
```
Comment (odinuv) [inline docker-compose.yml]: why? shouldn't that be a static value?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
     image: keboola/docker-jupyter
     tty: true
     stdin_open: true
-    volumes_from:
-      - data-loader
+    volumes:
+      - data:/data/
```
Comment (odinuv) [inline docker-compose.yml]: shouldn't this refer to the volume?

File: Dockerfile [hunk 5]
```diff
@@ -1,5 +1,12 @@
 FROM quay.io/keboola/docker-custom-python:1.5.11
 
+ARG NB_USER
+ARG NB_UID="1000"
+ARG NB_GID="100"
+ENV NB_USER=$NB_USER
```
Comment (odinuv) [inline Dockerfile]: > line 59

File: .travis.yml [hunk 6]
```diff
@@ -3,7 +3,7 @@ language: bash
 services:
   - docker
 before_script:
-  - docker build --tag keboola/docker-jupyter .
+  - docker-compose build
```
Comment (odinuv) [inline .travis.yml]: why?

Comment (pivnicek) [reply .travis.yml]: I changed to the compose to support the arg, but yeah, that's not necessary if it'll be static

File: docker-compose.yml (hunk 4 above)
Comment (pivnicek) [reply docker-compose.yml]: how do you mean?

File: docker-compose.yml (hunk 3 above)
Comment (pivnicek) [reply docker-compose.yml]: yes, it was something I was trying, but it doesn't make sense, thanks.

File: Dockerfile (hunk 1 above)
Comment (pivnicek) [reply Dockerfile]: yes, it is run as root, the `--user` option is left-over from a previous attempt, I'll remove it, thanks

File: Dockerfile (hunk 2 above)
Comment (pivnicek) [reply Dockerfile]: deleted

File: docker-compose.yml (hunk 4 above)
Comment (odinuv) [reply docker-compose.yml]: solved over phone

**Review by odinuv — APPROVED** [review]

**Comment by pivnicek** [comment]
Sorry @odinuv, while working on julipyter I noticed a couple of things that were wrong.

**Review by odinuv — APPROVED** [review]

**Pull request merged**